File: visiteur-api/src/main/java/com/visiteur/visiteur_api/model/Visiteur.java
package com.visiteur.visiteur_api.model;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import lombok.Data;

@Entity
@Table(name = "visiteur")
@Data
public class Visiteur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long numvisiteur;

    @Column(nullable = false)
    private String nom;

    @Column(name = "nombre_jours", nullable = false)
    private Integer nombreJours;

    @Column(name = "tarif_journalier", nullable = false)
    private BigDecimal tarifJournalier;

    @Transient
    public BigDecimal getTarif() {
        if (nombreJours != null && tarifJournalier != null) {
            return tarifJournalier.multiply(BigDecimal.valueOf(nombreJours));
        }
        return BigDecimal.ZERO;
    }
}

File: VisiteurApp/app/src/main/java/com/visiteur/app/model/Visiteur.kt
package com.visiteur.app.model

import com.google.gson.annotations.SerializedName

data class Visiteur(
    @SerializedName("numvisiteur")
    val numvisiteur: Long = 0,

    @SerializedName("nom")
    val nom: String = "",

    @SerializedName("nombreJours")
    val nombreJours: Int = 0,

    @SerializedName("tarifJournalier")
    val tarifJournalier: Double = 0.0
) {
    val tarif: Double
        get() = nombreJours * tarifJournalier
}